### PR TITLE
[DEVTOOLS-485] WordPress Project Setup for Forum One & Pantheon Hosting

### PR DESCRIPTION
**Pull request by timnolte**
* Adds Buildkite pipeline configuration for Forum One & Pantheon hosting.
* Adds Buildkite Docker Compose configuration for builds.
* Updates Composer configuration with base set of starter plugins & package dependencies.
* Adds code quality linting support for PHP Code Sniffer & PHP Mess Detector.
* Adds code formatting EditorConfig configuration.
* Adds Capistrano build configuration for Forum One hosting.
* Adds Pantheon hosting support configuration.
* Adds DDev starting configuration.
  * Adds custom Gesso theme service and supporting commands.
  * Adds custom Redis service and support commands.
  * Adds code quality check commands.
  * Adds Pantheon hosting provider support configuration.
* Adds starter webroot directory with WP-CFM bundles & ACF JSON configuration support.
* Adds universal hosting support WP Starter setup for WordPress configuration.
* Adds universal hosting support for Must Use Plugins for ACF & WP-CFM.
* Updates `.gitignore` & `.dockerignore` for DDev & Buildkite builds.
* Changes repository `README.md` to be the build documentation with instruction for Forum One and Pantheon hosting.
* Moves the project `README.md` template to `README.project.md`
* Adds a default `robots.txt` for a WordPress site.

### CODE REVIEW

File: .buildkite/pipeline.f1.yml [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+          from: /var/www/html
+          to: ./
+
+      # Execute a Capistrano deployment using the defined branch to
+      # environment mapping.
+      - forumone/capistrano:
+          require-stage: true
+          branches:
+            prerelease: prerelease
+            release: release
```
Comment (slucero) [inline .buildkite/pipeline.f1.yml:153]: These branches aren't included in the step filters so they'll never be deployed with Capistrano.

File: .buildkite/pipeline.pantheon.yml [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+      # Log into ECR for this build step to access and push images.
+      - *ecr
+      # Extract artifacts from the release image.
+      - forumone/extract#v0.2.0:
+          image:
+            service: release
+          from: /var/www/html
+          to: ./
+      # Push artifacts to Git remote.
+      - forumone/artifact-push#v0.3.0:
```
Comment (slucero) [inline .buildkite/pipeline.pantheon.yml:148]:
```suggestion
      - forumone/artifact-push#v0.3.1:
```

File: Dockerfile [hunk 3]
```diff
@@ -69,7 +69,7 @@ FROM release AS test
 
 COPY --from=composer-dev ["/app/web", "web"]
 
-COPY --from=gesso-dev ["/app", "web/wp-content/themes/gesso"]
+COPY --from=gesso-dev ["/app", "web  /wp-content/themes/gesso"]
```
Comment (slucero) [inline Dockerfile:72]:
```suggestion
COPY --from=gesso-dev ["/app", "web/wp-content/themes/gesso"]
```

File: README.md [hunk 4]
```diff
@@ -1,62 +1,200 @@
...
+### Pantheon hosted project,
+
+* Rename `.buildkite/pipeline.pantheon.yml` to `.buildkite/pipeline.yml`.
+  * Modify the following variables:
+    * `.buildkite/pipeline.yml` > `[SSH_REPO_FROM_PANTHEON]`
+    * `.buildkite/pipeline.yml` > `[PANTHEON_HOST]`
+    * `.ddev/providers/pantheon.yaml` > `[PROJECT.KEY]`, see the instructions in the `pantheon.yml` file.
+* Delete `.buildkite/pipeline.f1.yml`.
+* Delete the `web/wp-content/config` directory.
+* Delete the `capistrano` directory.
```
Comment (slucero) [inline README.md:108]:
```suggestion
* Delete the `capistrano` directory.
* Delete the `Capfile` file
* Delete the `Gemfile` file
```

While adding this, I also noticed the `Gemfile` isn't present at the root of the repo which I believe will cause the Capistrano steps to fail.

**Review by slucero — CHANGES_REQUESTED** [review]

**Comment by timnolte** [comment]
@slucero requested changes made. I also added 3 additional Pantheon specific WordPress plugins that we should be using and updated the README to speak to those in terms of removing for F1 Hosting. Thanks!

**Pull request merged**